arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -0,0 +1,265 @@
+        syncGeodatabaseJob.start(statusHandler: { (status: AGSJobStatus) in
+            SVProgressHUD.show(withStatus: status.statusString())
+        },
+                                 completion: { (result: [AGSSyncLayerResult]?, error: Error?) in
+                                    SVProgressHUD.dismiss()
+                                    if let error = error {
+                                        self.presentAlert(error: error)
+                                    } else {
+                                            self.presentAlert(title: "Geodatabase sync sucessful")
+                                            self.syncButton.isEnabled = false
+                                            self.syncButton.isHidden = true
+                                            self.tapFeatureLabel.isHidden = false
+                                        }
+                                    }
+        )
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["EditAndSyncFeaturesViewController"]
+    }
+}
+
+// Allows the user to interactively select and move features on the map.
+extension EditAndSyncFeaturesViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Move the selected feature to the tapped location and update it in the feature table.
+        if self.selectedFeature != nil {
+            let point = mapView.screen(toLocation: screenPoint)
+            if AGSGeometryEngine.geometry(point, intersects: areaOfInterest) {
+                selectedFeature.geometry = point
+                selectedFeature.featureTable?.update(selectedFeature) { [weak self] (error: Error?) in
+                    if let error = error {
+                        self!.presentAlert(error: error)
+                    } else {
+                        self!.syncButton.isHidden = false
+                        self!.syncButton.isEnabled = true
+                        self!.tapFeatureLabel.isHidden = true
+                        self!.moveFeatureLabel.isHidden = true
+                        self!.generateButton.isHidden = true
+                        self!.tapSyncLabel.isHidden = false
+                    }
+                }
+            } else {
+                self.presentAlert(title: "Cannot move feature outside downloaded area")
+            }
+        } else { // Identify which feature was tapped and select it.
+            mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 22.0, returnPopupsOnly: false, maximumResultsPerLayer: 1) { (results: [AGSIdentifyLayerResult]?, error: Error?) in
+                if let error = error {
+                    self.presentAlert(error: error)
+                } else {
```

Comment:
I would suggest optionally unwrapping `results` here so as not to force unwrap it later:

```swift
                } else if let results = results {
```